dpe/src/context.rs

```diff
@@ -150,70 +150,173 @@ pub(crate) struct ChildToRootIter<'a> {
     idx: usize,
     contexts: &'a [Context],
     done: bool,
+    count: usize,
 }
 
 impl ChildToRootIter<'_> {
     /// Create a new iterator that will start at the leaf and go to the root node.
-    pub fn new(leaf_idx: usize, contexts: &[Context]) -> Result<ChildToRootIter, DpeErrorCode> {
-        Ok(ChildToRootIter {
+    pub fn new(leaf_idx: usize, contexts: &[Context]) -> ChildToRootIter {
+        ChildToRootIter {
             idx: leaf_idx,
             contexts,
             done: false,
-        })
+            count: 0,
+        }
     }
 }
 
 impl<'a> Iterator for ChildToRootIter<'a> {
-    type Item = &'a Context;
+    type Item = Result<&'a Context, DpeErrorCode>;
 
-    fn next(&mut self) -> Option<&'a Context> {
+    fn next(&mut self) -> Option<Result<&'a Context, DpeErrorCode>> {
         if self.done {
             return None;
         }
+        if self.count >= MAX_HANDLES {
+            return Some(Err(DpeErrorCode::MaxTcis));
```

Comment:
Which semantics do we want? Currently we repeatedly and indefinitely return the same error. An alternative would be to `self.done = false` before returning the first error, such that iteration stops after the first error.

It looks to me that finite iteration is "safer" in the sense that one cannot inadvertently "freeze" execution because they forgot to interrupt iteration on error.

Reply:
No I mean marking the iterator as done such that if the user continues to call next after an error, then iteration would stop instead of infinitely returning the same error which would most probably mean an infinite loop.

Reply:
Ahhh, yeah, got it. That is a good idea! I just added it.